src/resources/triton_python_backend_utils.py

```diff
@@ -340,6 +342,23 @@ def set_max_batch_size(self, max_batch_size):
                 str(max_batch_size))
         else:
             self._model_config["max_batch_size"] = max_batch_size
+            if max_batch_size > 1:
+                self.set_dynamic_batching()
+
+    def set_dynamic_batching(self):
+        """Set dynamic_batching as the scheduler for the model if no scheduler is set.
+           If dynamic_batching is already set, then no action is taken and return success.
+        Raises
+        ------
+        ValueError
+            If the configuration has a scheduler set which is not dynamic_batching.
```

Comment:
I don't see ValueError being raised anywhere. Maybe we can raise a pb_utils.TritonModelException to be consistent with other APIs when sequence_batching/ensemble_scheduling/dynamic_batching exists in the configuration?